target_chains/ton/contracts/contracts/imports/utils.fc

```diff
@@ -0,0 +1,75 @@
+;; Built-in assembly functions
+int keccak(slice s, int h) asm "HASHEXT_KECCAK256"; ;; Keccak-256 hash function
+int keccak_4(slice s1, slice s2, slice s3, slice s4, int h) asm "HASHEXT_KECCAK256";  ;; Keccak-256 hash function for vm body since it spans across 4 slices
+
+int keccak256(slice s) inline {
+    return keccak(s, 1);
+}
+
+int hash_vm_body(slice s) inline {
+    slice s1 = s~load_bits(s.slice_bits());
+    s = s~load_ref().begin_parse();
+    slice s2 = s~load_bits(s.slice_bits());
+    s = s~load_ref().begin_parse();
+    slice s3 = s~load_bits(s.slice_bits());
+    s = s~load_ref().begin_parse();
+    slice s4 = s~load_bits(s.slice_bits());
+    int hash = keccak_4(s1, s2, s3, s4, 4);
+    slice hash_slice = begin_cell().store_uint(hash, 256).end_cell().begin_parse();
+    return keccak256(hash_slice);
+}
+
+;; Splits a slice into chunks of 1016 bits or less, in reverse order
+(cell, slice) split_into_reverse_chunks(slice data, int size) {
+    cell chunks = null();
+    int total_bits_loaded = 0;
+    builder current_chunk = begin_cell();
+    while ((~ data.slice_empty?()) & (total_bits_loaded < size)) {
```

Comment:
hmm I agree that this might be unclear to non-TON readers but I believe that using the ~ operator is more idiomatic in TON development, since our primary audience for this particular contract consists of TON developers, it might be beneficial to stick with the native syntax